drivers/sensor/icm42670S/icm42670S_spi.c

```diff
@@ -21,46 +21,40 @@ static int icm42670S_bus_check_spi(const union icm42670S_bus *bus)
 }
 
 static int icm42670S_reg_read_spi(const union icm42670S_bus *bus, uint8_t start, uint8_t *buf,
-				  int size)
+				  uint32_t size)
 {
-	uint8_t addr;
-	const struct spi_buf tx_buf = {.buf = &addr, .len = 1};
+	uint8_t cmd[] = {(start | 0x80), 0};
+	const struct spi_buf tx_buf = {.buf = cmd, .len = sizeof(cmd)};
 	const struct spi_buf_set tx = {.buffers = &tx_buf, .count = 1};
 	struct spi_buf rx_buf[2];
 	const struct spi_buf_set rx = {.buffers = rx_buf, .count = ARRAY_SIZE(rx_buf)};
-	int i;
+	int ret;
 
 	rx_buf[0].buf = NULL;
 	rx_buf[0].len = 1;
 
-	rx_buf[1].len = 1;
-
-	for (i = 0; i < size; i++) {
-		int ret;
-
-		addr = (start + i) | 0x80;
-		rx_buf[1].buf = &buf[i];
+	rx_buf[1].len = size;
+	rx_buf[1].buf = buf;
 
-		ret = spi_transceive_dt(&bus->spi, &tx, &rx);
-		if (ret) {
-			LOG_DBG("spi_transceive FAIL %d\n", ret);
-			return ret;
-		}
+	ret = spi_transceive_dt(&bus->spi, &tx, &rx);
+	if (ret) {
+		LOG_ERR("spi_transceive FAIL %d\n", ret);
+		return ret;
 	}
-
 	return 0;
 }
 
-static int icm42670S_reg_write_spi(const union icm42670S_bus *bus, uint8_t reg, uint8_t val)
+static int icm42670S_reg_write_spi(const union icm42670S_bus *bus, uint8_t reg, uint8_t *buf,
+				   uint32_t size)
 {
-	uint8_t cmd[] = {reg & 0x7F, val};
+	uint8_t cmd[] = {reg & 0x7F, *buf};
 	const struct spi_buf tx_buf = {.buf = cmd, .len = sizeof(cmd)};
-	const struct spi_buf_set tx = {.buffers = &tx_buf, .count = 1};
+	const struct spi_buf_set tx = {.buffers = &tx_buf, .count = size};
```

Comment:
I'm pretty sure ".count=size" should be rpelaced by ".count=1".
It probably works because buf length is always 1, but the behavior won't be correct if the driver tries to write 2 bytes!

Reply:
Ok, I modified this.